src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -925,8 +926,11 @@ private async Task<bool> IsPackagesConfigBasedProjectAsync()
             return true;
         }
 
-
-        private bool IsSolutionOpen => _dte?.Solution != null && _dte.Solution.IsOpen;
+        private bool GetIsSolutionOpen()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            return _dte?.Solution != null && _dte.Solution.IsOpen;
```

Comment:
Why this check needs to the on the UI thread? Maybe add comments for the reason?

Reply:
dte objects have thread affinity, the analyzers catch this. 

https://github.com/microsoft/vs-threading/blob/master/doc/threading_rules.md#rule-1-if-a-method-has-certain-thread-apartment-requirements-sta-or-mta-it-must-either

It's a common pattern in Visual Studio code, so we don't add comments why explicitly.